homeassistant/components/switch/mqtt.py

```diff
@@ -62,6 +62,7 @@
     async def async_discover(discovery_payload):
         """Discover and add a MQTT switch."""
         config = PLATFORM_SCHEMA(discovery_payload)
+        config['config_entry'] = config_entry
```

Comment:
I don't like this line too much. `config` should ideally just contain the user config (and should be JSON serializable). Could we make this an additional parameter to the constructor like `discovery_hash`?

Reply:
OK, will fix!